### DISCUSSION TITLE
Patching `AddTool.patch` file

### DISCUSSION THREAD

**Discussion by ApeachM** [The-OpenROAD-Project/OpenROAD#3735] [Q&A] [answered]
Hello.

Looking at the document in `docs/contrib/DeveloperGuide.md`,  following the `Example of Adding a Tool to OpenROAD` section, I would like to patch [`docs/misc/AddTool.patch`](https://github.com/The-OpenROAD-Project/OpenROAD/blob/master/docs/misc/AddTool.patch) file but the error message has occurred. 

```shell
minjae@minjae-desktop:~/OpenROAD3DFlow/docs/misc$ patch -p < AddTool.patch
patch: option requires an argument -- 'p'
patch: Try 'patch --help' for more information.
```

I am not familiar with patch files, so I can't assume what is the problem.

If you let me know then I would be very appreciated.

Thank you.

**Comment by maliberty** [comment]
Hi

Please see
https://openroad.readthedocs.io/en/latest/contrib/DeveloperGuide.html#example-of-adding-a-tool-to-openroad

The patch file itself is a bit old so some updating might be needed.  What
new tool are you adding?

Matt

**Reply by ApeachM** [reply]
I would like to add a tool for `f3d`, which can handle 3D IC design. 
I followed the link that you gave me, but the below command
```shell
patch -p < AddTool.patch
```
occurred the error below.
```shell
patch: option requires an argument -- 'p'
patch: Try 'patch --help' for more information
```

I can't assure you that it is a problem with me or the OpenROAD repository.

Thanks for answering!

**Reply by maliberty** [reply, marked as answer]
patch -p1 < docs/misc/AddTool.patch

However I do see some errors.  This example is quite old and in need of updating

@vvbandeira perhaps this can be addressed as part of the docs work.

**Reply by ApeachM** [reply]
Thank you for your answer, Matt!
